test/runtests.jl

```diff
@@ -34,3 +34,10 @@ println("Testing miscellaneous functionality...")
 tic()
 include("MiscTest.jl")
 println("done (took $(toq()) seconds).")
+
+if Base.JLOptions().opt_level >= 3 && VERSION >= v"0.5"
```

Comment:
Alright, attempted what you suggested. Let's see if Travis's infrastructure works for this...